Sources/PackageCollectionsSigning/CertificatePolicy.swift

```diff
@@ -0,0 +1,644 @@
+            return "Swift Package Collection certificate policy\(userID.map { " (userID: \($0))" } ?? "")\(organizationalUnit.map { " (organizationalUnit: \($0))" } ?? "")"
+        case .appleDistribution(let userID, let organizationalUnit):
+            return "Distribution certificate policy\(userID.map { " (userID: \($0))" } ?? "")\(organizationalUnit.map { " (organizationalUnit: \($0))" } ?? "")"
+        case .custom:
+            return "Custom certificate policy"
+        }
+    }
+
+    public static let `default` = CertificatePolicyKey.default()
+    public static let appleSwiftPackageCollection = CertificatePolicyKey.appleSwiftPackageCollection()
+    @available(*, deprecated, message: "use `appleSwiftPackageCollection` instead")
+    public static let appleDistribution = CertificatePolicyKey.appleDistribution()
+}
+
+// MARK: - Certificate policies
+
+protocol CertificatePolicy {
+    /// Validates the given certificate chain.
+    ///
+    /// - Parameters:
+    ///   - certChainPaths: Paths to each certificate in the chain. The certificate being verified must be the first
+    ///                     element of the array, with its issuer the next element and so on, and the root CA
+    ///                     certificate is last.
+    ///   - validationTime: Overrides the timestamp used for checking certificate expiry (e.g., for testing).
+    ///                     By default the current time is used.
+    ///   - callback: The callback to invoke when the result is available.
+    func validate(certChain: [Certificate], validationTime: Date, callback: @escaping (Result<Void, Error>) -> Void)
+}
+
+extension CertificatePolicy {
+    /// Validates the given certificate chain.
+    ///
+    /// - Parameters:
+    ///   - certChainPaths: Paths to each certificate in the chain. The certificate being verified must be the first
+    ///                     element of the array, with its issuer the next element and so on, and the root CA
+    ///                     certificate is last.
+    ///   - callback: The callback to invoke when the result is available.
+    func validate(certChain: [Certificate], callback: @escaping (Result<Void, Error>) -> Void) {
+        self.validate(certChain: certChain, validationTime: Date(), callback: callback)
+    }
+
+    func verify(
+        certChain: [Certificate],
+        trustedRoots: [Certificate]?,
+        policies: [VerifierPolicy],
+        observabilityScope: ObservabilityScope,
+        callbackQueue: DispatchQueue,
+        callback: @escaping (Result<Void, Error>) -> Void
+    ) {
+        let wrappedCallback: (Result<Void, Error>) -> Void = { result in callbackQueue.async { callback(result) } }
+
+        guard !certChain.isEmpty else {
+            return wrappedCallback(.failure(CertificatePolicyError.emptyCertChain))
+        }
+
+        Task {
+            var trustStore = CertificateStores.defaultTrustRoots
```

Comment:
A major change in the new implementation is that we no longer use the system trust store (we used to on Darwin platforms through the `Security` framework) and SwiftPM will its own default trust store instead.